Standardize authentication for omni_transferWithdraw and omni_exportWallet

silva-fj opened the pull request:
## Summary

Authentication Flow Updates
  - export_wallet and transfer_withdraw now use Authorization header authentication
  - All methods extract the OA from the user's JWT
  - Removed redundant user_email parameters

 Refactored protected RPC methods to extract AccountId from authenticated users and remove email parameter:
  - omni_exportWallet 
  - omni_transferWithdraw 
  - omni_addWallet 
  - omni_submitSwapOrder 
  - omni_signLimitOrder 
  - omni_notifyLimitOrderResult 

Refactored related NativeTask's to use OA instead of Identity.

Note: We need to refactor the omni-account pallet to adjust how intent calls are dispatched. We currently send the Identity but we should send the actual OA. Because of this, some code has been commented out (as well as the integration tests)

silva-fj committed:
refactor: update transfer_widthdraw to use Authorization header authentication

- Add omni_transferWithdraw to PROTECTED_METHODS array
- Replace parameter-based auth (email_code, client_id) with JWT token authentication
- Use check_auth() for consistent authentication handling across protected methods
- Remove manual verify_auth() calls in favor of middleware-based authentication
- Align authentication mechanism with other protected methods like submit_swap_order

silva-fj committed:
refactor: update export_wallet to use Authorization header authentication

- Add omni_exportWallet to PROTECTED_METHODS array
- Replace parameter-based auth (email_code, client_id) with JWT token authentication
- Use check_auth() for consistent authentication handling across protected methods
- Remove manual verify_auth() calls in favor of middleware-based authentication
- Align authentication mechanism with other protected methods

silva-fj committed:
Merge branch 'dev' into p-1583-use-generic-user-idauth-and-client-idauth-in

silva-fj committed:
fix fmt issue

silva-fj committed:
Merge branch 'dev' into p-1583-use-generic-user-idauth-and-client-idauth-in

Kailai-Wang commented on tee-worker/omni-executor/rpc-server/src/methods/omni/transfer_widthdraw.rs:
```diff
@@ -16,14 +17,12 @@ use super::common::{check_omni_api_response, handle_omni_native_task};
 #[derive(Debug, Deserialize)]
 pub struct TransferWithdrawParams {
 	pub user_email: String,
-	pub client_id: String,
```
There should be no `user_email` anymore

Kailai-Wang commented on tee-worker/omni-executor/rpc-server/src/methods/omni/export_wallet.rs:
```diff
@@ -17,18 +18,16 @@ use super::common::handle_omni_native_task;
 
 #[derive(Debug, Deserialize)]
 pub struct ExportWalletParams {
-	pub client_id: String,
```
There should be no `user_email` anymore

Kailai-Wang replied:
We just extract the OA from JWT if we need it. 
If we need the `user_email` to construct the `NativeTask` - then it's wrong, NativeTask shouldn't need the `Identity` anymore - as we verify the auth before processing `NativeTask`

silva-fj replied:
I thought about that but it probably requires a broader refactoring as all tasks are expected to have the sender's Identity. I'll try to make the changes here independently, then create a follow up PR if needed

silva-fj replied:
it turns out we were not using the sender method from the NativeTask so the change won't affect the other tasks that require the sender's identity

Kailai-Wang reviewed (changes requested):
Thanks - I think it requires a small refactoring of `NativeTask`

I assume checks against other methods would come in a follow-up PR?

silva-fj commented:
> I assume checks against other methods would come in a follow-up PR?

All other methods have been checked already. Some require no authentication and the others already use the Authorization header

silva-fj committed:
refactor: remove user_email param from omni_exportWallet and use AccountId in PumpxExportWallet

- Update NativeTask::PumpxExportWallet to use AccountId instead of Identity
- Remove user_email parameter from omni_exportWallet RPC method
- Extract omni account directly from authenticated user in RPC handlers
- Simplify native task handler by using AccountId directly instead of converting from Identity
- Update both omni and pumpx export wallet implementations for consistency

This simplifies the API by removing redundant parameters and improves security by using authenticated user data directly.

silva-fj committed:
refactor: remove user_email param from omni_transferWithdraw and use AccountId in PumpxTransferWidthdraw

- Update NativeTask::PumpxTransferWidthdraw to use AccountId instead of Identity
- Remove user_email parameter from omni_transferWithdraw RPC method
- Extract omni account directly from authenticated user in RPC handlers
- Simplify native task handler by using AccountId directly instead of converting from Identity
- Update both omni and pumpx transfer withdraw implementations for consistency

This change maintains consistency with the PumpxExportWallet refactoring and improves security by using authenticated user data directly.

silva-fj committed:
fix fmt issue

silva-fj committed:
Merge branch 'dev' into p-1583-use-generic-user-idauth-and-client-idauth-in

silva-fj committed:
feat: use AccountId for PumpxAddWallet native task

Replace Identity parameter with AccountId in PumpxAddWallet to simplify
authentication flow. This change:

- Updates NativeTask::PumpxAddWallet to accept AccountId directly
- Removes user_email parameter from omni add_wallet endpoint
- Extracts AccountId from authenticated user's omni_account token
- Converts Identity to omni_account in pumpx add_wallet endpoint
- Eliminates need for Identity conversion in native task handler

silva-fj committed:
refactor: use AccountId for PumpxSignLimitOrder task

Replace Identity parameter with AccountId in PumpxSignLimitOrder to
simplify parameter handling. This change:

- Updates NativeTask::PumpxSignLimitOrder to accept AccountId directly
- Removes Identity-to-AccountId conversion logic in task handler
- Updates both omni and pumpx RPC endpoints to pass AccountId
- Eliminates error handling for invalid Identity types

silva-fj committed:
refactor: use AccountId for PumpxNotifyLimitOrderResult

Replace Identity with AccountId parameter in PumpxNotifyLimitOrderResult
task to maintain consistency with other native tasks and simplify the
implementation by removing the Identity-to-AccountId conversion logic.

Changes:
- Update NativeTask enum to use AccountId instead of Identity
- Remove Identity conversion workaround in native task handler
- Update RPC method implementations to pass AccountId directly
- Clean up unused Identity imports

silva-fj committed:
refactor: use AccountId for RequestIntent native task

Replace Identity with AccountId parameter in RequestIntent task to
standardize parameter types across native tasks and eliminate redundant
Identity-to-AccountId conversions.

Changes:
- Update NativeTask::RequestIntent to accept AccountId directly
- Remove Identity conversion logic in native task handler
- Temporarily disable SystemRemark and TransferNative intents
- Update RPC endpoints to pass AccountId instead of Identity
- Clean up unused imports and commented code

silva-fj committed:
test: comment out RequestIntent test until refactor

Temporarily disable the RequestIntent TransferNative test while the
omni-account pallet is being refactored to use AccountId instead of
Identity parameters.

silva-fj committed:
Merge branch 'dev' into p-1583-use-generic-user-idauth-and-client-idauth-in

kziemianek commented on tee-worker/omni-executor/executor-core/src/native_task.rs:
```diff
@@ -41,7 +39,7 @@ pub type PumpxChainId = u32;
 	RequestAuthToken(Identity),
-	RequestIntent(Identity, IntentId, Intent),
+	RequestIntent(AccountId, IntentId, Intent),
 	CreateAccountStore(Identity),
```
Is it possible that remaining task types will accept `AccountId` instead of `Identity` ?

Kailai-Wang replied:
@kziemianek do you mean old tasks (non-dex-related)? If so we could do it in another PR - maybe together with AccountStore cleanup 🤔

kziemianek replied:
Yes, ok

Kailai-Wang commented on tee-worker/omni-executor/native-task-handler/src/lib.rs:
```diff
@@ -1030,33 +1012,33 @@ fn send_ok(sender: ResponseSender, ok_res: NativeTaskOk) {
-	auth_type: Option<OmniAccountAuthType>,
-) {
-	let call = parentchain_api_interface::tx().omni_account().dispatch_as_signed(
-		sender.hash().to_subxt_type(),
```
IIUC the problem here is the `hash()` right? We could later use the hash of the OmniAccount - either blake2_256 or another sha2 if we want it to be evm-friendly.

Can do together with AccountStore cleanup

Kailai-Wang reviewed (approved):
Looks good, thanks.

Could you (or ask Verin) to make sure the apifox is up-to-date?

Kailai-Wang committed:
Merge branch 'dev' into p-1583-use-generic-user-idauth-and-client-idauth-in
